web/src/ui/pages/projectSettings/ProjectSettingsS3ConfigTab/S3ConfigCard.tsx

```diff
@@ -80,7 +82,16 @@ export function S3ConfigCard(props: Props) {
                 </Tooltip>
                 &nbsp;
                 <Switch
-                    checked={s3Config.isXOnyxiaDefault}
+                    disabled={
+                        canInjectPersonalInfos
+                            ? false
+                            : s3Config.origin === "deploymentRegion"
+                    }
+                    checked={
+                        canInjectPersonalInfos || s3Config.origin !== "deploymentRegion"
+                            ? s3Config.isXOnyxiaDefault
+                            : false
+                    }
```

Comment:
This logic should ideally be handled by the core. The component shouldn't need to be so aware of the intricacies of S3Config management; its responsibility should be limited to rendering the UI and forwarding the relevant user input.

That said, I understand the desire to avoid introducing another type just to adhere strictly to the doctrine, so I'll approve this impl.